### PR TITLE
Moving to rsync

### PR DESCRIPTION
**Pull request by tolzhabayev**
Following up on: https://github.com/grafana/plugin-tools/pull/1214 and https://github.com/grafana/grafana-developer-portal/pull/22

Switching to using rsync with delete option - to ensure that the GCS content equals the content of the build, up until now we just added new files but never deleted the old ones.  Since it is a static site - it can lead to some weird old built pages still being available even when deleted from the source code. This PR cleans it up.

Starting as dry run `-n` and will remove it before merge.

Additional yak shaving changes:
- Bump docusaurus to the v3
- Fix bump-related issues in mdx file in advance variables doc
- Adjust turborepo build to be able to pass arguments required for dev stage build
- Move towards cleaner config - with shared underlying base file

Example runs:
- Dev https://github.com/grafana/scenes/actions/runs/11343394573/job/31545849078
-  Prod - will run once merged

- [x] Remove dry run for dev
- [x] Remove dry run for prod

### CODE REVIEW

File: docusaurus/website/docusaurus.config.base.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
+const path = require('path');
```
Comment (jackw) [inline docusaurus/website/docusaurus.config.base.js:1]: These config changes might require bumping scenes docusaurus to v3. 🤔

File: docusaurus/website/docusaurus.config.base.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+const presetsDocs = {
+  path: '../docs',
+  sidebarPath: require.resolve('./sidebars.js'),
+  // Please change this to your repo.
+  // Remove this to remove the "edit this page" links.
+  editUrl: 'https://github.com/grafana/scenes/edit/main/docusaurus/website',
+};
+
+const presetsTheme = {
+  customCss: require.resolve('./src/css/custom.css'),
```
Comment (jackw) [inline docusaurus/website/docusaurus.config.base.js:72]: The css files might need an update too.

File: docusaurus/website/docusaurus.config.base.js (hunk 1 above)
Comment (tolzhabayev) [reply docusaurus/website/docusaurus.config.base.js:1]: Bumped https://github.com/grafana/scenes/pull/938/commits/a50c20a7050df4c16f40937bf624662e3ef7f554

File: docusaurus/website/docusaurus.config.base.js (hunk 2 above)
Comment (tolzhabayev) [reply docusaurus/website/docusaurus.config.base.js:72]: Updated https://github.com/grafana/scenes/pull/938/commits/c736f89f57043ca0c06448b4286b583656024321

File: docusaurus/website/docusaurus.config.dev.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+const devPortalHome = 'https://grafana-dev.com/developers';
+const [docsFooterLinks, ...otherFooterLinks] = themeConfigFooter.links;
+
+/** @type {import('@docusaurus/types').Config} */
+const config = {
+  ...generalConfig,
+  url: 'https://grafana-dev.com/',
+  baseUrl: 'developers/scenes/',
+  plugins: [],
```
[the remark this answers is not in the document]
Comment (tolzhabayev) [reply docusaurus/website/docusaurus.config.dev.js]: thanks, fixed

**Comment by tolzhabayev** [comment]
@dprokop I managed to fix the build issues related to cache by removing the single github actions caches manually

`gh cache list`

but I think the cache logic you are using in this workflow should be re-evaluated https://github.com/grafana/scenes/blob/main/.github/actions/yarn-nm-install/action.yml as it will most likely happen to other PRs.

**Pull request merged**